### PR TITLE
Fix issuer tests

### PR DESCRIPTION
**Pull request by PatStLouis**
Some of the work from #82 got lost during re-base, I'm re-introducing it here. This should have already been reviewed but please have a look!

### CODE REVIEW

File: tests/10-vcdm2.js [hunk 1]
```diff
@@ -417,27 +417,28 @@ describe('Issuer', function() {
...
         await endpoints.issue(require(
           './input/credential-issuer-object-ok.json'));
-        await endpoints.issue(require(
-          './input/credential-issuer-url-ok.json')
-        );
         await assert.rejects(endpoints.issue(require(
-          './input/credential-issuer-nonurl-fail.json')),
-        'Failed to reject a numeric issuer identifier.');
+          './input/credential-issuer-no-url-fail.json')),
+        'Failed to reject a non url issuer identifier.');
```
Comment (TallTed) [inline tests/10-vcdm2.js:432]:
```suggestion
        'Failed to reject an issuer identifier that was not a URL.');
```

File: tests/10-vcdm2.js [hunk 2]
```diff
@@ -417,27 +417,28 @@ describe('Issuer', function() {
...
+        await assert.rejects(endpoints.issue(require(
+          './input/credential-issuer-null-fail.json')),
+        'Failed to reject an issuer null identifier.');
```
Comment (TallTed) [inline tests/10-vcdm2.js:435]:
```suggestion
        'Failed to reject a null issuer identifier.');
```

File: tests/10-vcdm2.js [hunk 3]
```diff
@@ -417,27 +417,28 @@ describe('Issuer', function() {
...
         await assert.rejects(endpoints.issue(require(
-          './input/credential-issuer-object-no-id-fail.json')),
-        'Failed to reject an issuer with an empty object as the value.');
+          './input/credential-issuer-object-id-null-fail.json')),
+        'Failed to reject an issuer object containing a null identifier.');
         await assert.rejects(endpoints.issue(require(
-          './input/credential-issuer-object-id-not-url-fail.json')),
-        'Failed to reject a issuer object containing a numeric identifier.');
+          './input/credential-issuer-object-id-no-url-fail.json')),
+        'Failed to reject a issuer object containing a non url identifier.');
```
Comment (TallTed) [inline tests/10-vcdm2.js:441]:
```suggestion
        'Failed to reject an issuer object containing a non URL identifier.');
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Grammar fixes

File: tests/mock.data.js [hunk 4]
```diff
@@ -14,9 +14,13 @@ export const createRequestBody = ({issuer, vc = validVc}) => {
   const {settings: {id, options}} = issuer;
   const credential = klona(vc);
   if(typeof credential.issuer === 'object') {
-    credential.issuer.id = credential.issuer?.id || id;
+    if('id' in credential.issuer && credential.issuer.id === null) {} else {
+      credential.issuer.id = credential.issuer?.id || id;
+    }
```
[the remark this answers is not in the document]
Comment (BigBlueHat) [reply tests/mock.data.js:19]: Definitely best to make the condition negative and avoid the empty positive test and "hidden" `else`. Good suggestion, thanks @tminard.

[the remark this answers is not in the document]
Comment (PatStLouis) [reply tests/mock.data.js:19]: let me give that a try and report back, if I get the same results I'm happy with this suggestion.

**Review by aljones15 — APPROVED** [review]
thanks for the catch and fixing this.

File: tests/10-vcdm2.js [hunk 5]
```diff
@@ -417,27 +417,28 @@ describe('Issuer', function() {
...
         await assert.rejects(endpoints.issue(require(
-          './input/credential-issuer-object-no-id-fail.json')),
-        'Failed to reject an issuer with an empty object as the value.');
+          './input/credential-issuer-object-id-null-fail.json')),
+        'Failed to reject an issuer object containing a null identifier.');
         await assert.rejects(endpoints.issue(require(
-          './input/credential-issuer-object-id-not-url-fail.json')),
-        'Failed to reject a issuer object containing a numeric identifier.');
+          './input/credential-issuer-object-id-no-url-fail.json')),
+        'Failed to reject an issuer object containing a non URL identifier.');
```
Comment (TallTed) [inline tests/10-vcdm2.js:441]:
```suggestion
        'Failed to reject an issuer object containing a non-URL identifier.');
```

**Comment by PatStLouis** [comment]
@tminard Let me know if I got your idea right

File: tests/TestEndpoints.js [hunk 6]
```diff
@@ -26,8 +26,7 @@ export class TestEndpoints {
   async issue(credential) {
     const {issuer} = this;
     const issueBody = createRequestBody({issuer, vc: credential});
-    const response = await post(issuer, issueBody);
-    return response?.verifiableCredential || response;
```
Comment (BigBlueHat) [inline tests/TestEndpoints.js:30]: Curious what caused this change. Does this never happen now?

Comment (PatStLouis) [reply tests/TestEndpoints.js:30]: @aljones15 pointed out [this section to me](https://github.com/w3c/vc-test-suite-implementations/blob/056b8d6e00275ce13656a372342c770125ef9264/lib/requests.js#L115-L124). I tested it again to confirm and I can't reproduce the behavior I had before

**Review by TallTed — APPROVED** [review]

File: tests/TestEndpoints.js (hunk 6 above)
Comment (BigBlueHat) [reply tests/TestEndpoints.js:30]: So, I guess the "format is stabilized" enough to remove it then. Hooray for simplicity. 😄

**Pull request merged**